Applications/web_portal/web_app/library/source/pages/login_page/login_page.tsx

```diff
@@ -11,15 +15,220 @@ export interface Properties {
+                data='resources/login_page/logo-static.svg'
+                type='image/svg+xml' className={
+                css(LoginPage.STYLE.logoVisible)}/>
+              <object id='animated-logo'
+                data='resources/login_page/logo-animated.svg'
+                type='image/svg+xml' className={
+                css(LoginPage.STYLE.logoInvisible)}>
+              </object>
+              <Padding/>
+           </HBoxLayout>
+            <Padding size='60px'/>
+            <input type='text' placeholder='Username' autoComplete='off'
+              className={css(LoginPage.STYLE.inputBox)}
+              onChange={() => this.onInputChange()}
+              onFocus={() => this.usernameInputField.placeholder = ''}
+              onBlur={() => this.usernameInputField.placeholder = 'Username'}
+              ref={((ref) => this.usernameInputField = ref)}/>
+            <Padding size='20px'/>
+            <input type='password' placeholder='Password' autoComplete='off'
+              className={css(LoginPage.STYLE.inputBox)}
+              onChange={() => this.onInputChange()}
+              onFocus={() => this.passwordInputField.placeholder = ''}
+              onBlur={() => this.passwordInputField.placeholder = 'Password'}
+              ref={((ref) => this.passwordInputField = ref)}/>
+            <Padding size='50px'/>
+            <button className={css(LoginPage.STYLE.signInButton)}
+                onClick={this.onLogin}
+                ref={((ref) => this.submitButton = ref)}>
+              Sign In
+            </button>
+            <Padding size='30px'/>
+            <span className={css(LoginPage.STYLE.errorMessage)}>
+              {this.state.errorMessage}
+            </span>
+            <Padding size='60px'/>
+          </VBoxLayout>
+          <Padding size='18px'/>
+        </HBoxLayout>
+      </Center>);
+  }
+
+  private async onLogin() {
+    if(this.usernameInputField.value.trim() !== '') {
+      const staticLogo = document.getElementById('static-logo');
+      const animatedLogo = document.getElementById('animated-logo');
+      staticLogo.className = css(LoginPage.STYLE.logoInvisible);
+      animatedLogo.className = css(LoginPage.STYLE.logoVisible);
+      try {
+        const account = await this.props.model.login(
+          this.usernameInputField.value, this.passwordInputField.value);
+        staticLogo.className = css(LoginPage.STYLE.logoVisible);
+        animatedLogo.className = css(LoginPage.STYLE.logoInvisible);
+        this.setState({
+          errorMessage: null
+        });
+        if(this.props.onLogin) {
+          this.props.onLogin(account);
+        }
+      } catch(error) {
+        staticLogo.className = css(LoginPage.STYLE.logoVisible);
+        animatedLogo.className = css(LoginPage.STYLE.logoInvisible);
+        this.setState({
+          errorMessage: error.toString()
+        });
+      }
+    }
+  }
+
+  private onInputChange() {
+    if(this.state.errorMessage) {
```

Comment:
Because errorMessage is a nullable string and JavaScript is wonky, this check is too general. It will be true if it's null, or an empty string when you want to check for null. As such use a strict check of this.state.errorMessage === null